src/main/java/org/jenkinsci/plugins/jx/resources/ProwEnvironmentContributor.java

```diff
@@ -15,9 +17,21 @@
 public class ProwEnvironmentContributor extends EnvironmentContributor {
     @Override
     public void buildEnvironmentFor(@Nonnull Run r, @Nonnull EnvVars envs, @Nonnull TaskListener listener) {
-        if (System.getenv("PROW_JOB_ID") != null) {
-            envs.put("BUILD_ID", System.getenv("BUILD_ID"));
-            envs.put("BUILD_NUMBER", System.getenv("BUILD_NUMBER"));
+        String prowJobId = getEnv("PROW_JOB_ID");
+        if (prowJobId != null) {
+            String buildId = getEnv("BUILD_ID");
+            if (buildId != null) {
+                // Overrides Jenkins predefined environment variables with the one provided by Prow
+                envs.put("BUILD_ID", buildId);
```

Comment:
Not sure I get it. Why would you override `$BUILD_ID` with the same value?

Reply:
In this context (serverless Jenkins), the `BUILD_ID` passed by Jenkins to the build would always be `1`. This takes the `BUILD_ID` set by Prow (https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables) to Jenkins Master environment and pass it down to the build context.